src/transformers/__init__.py

```diff
@@ -710,7 +710,7 @@
 
 # Tensorflow-text-specific objects
 try:
-    if not is_tensorflow_text_available():
+    if not (is_tensorflow_text_available() and is_tf_available()):
```

Comment:
I don't like this change much, but the import of `TFBertTokenizer` and `TFGPT2Tokenizer` also requires `tensorflow` too.

cc @Rocketknight1